core/trino-main/src/test/java/io/trino/sql/planner/TestDynamicFilter.java

```diff
@@ -118,6 +119,24 @@ public void testRightEquiJoin()
                                                 tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey")))))));
     }
 
+    @Test
+    public void testRightEquiJoinWithLeftExpression()
+    {
+        assertPlan("SELECT o.orderkey FROM orders o RIGHT JOIN lineitem l ON l.orderkey + 1 = o.orderkey",
+                output(
+                        join(
+                                RIGHT,
+                                ImmutableList.of(equiJoinClause("ORDERS_OK", "expr")),
+                                ImmutableMap.of("ORDERS_OK", "expr"),
+                                anyTree(
+                                        tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
+                                exchange(
+                                        project(
```

Comment:
what is this project (I think it's hash generator)?

Use `anyTree` (instead of `exchange` and `project`) to make the test more resilient

Reply:
> Yup, it's hash. anyTree used.

I don't think you pushed that